### PR TITLE
Add product pricing block

### LINKED ISSUE #37098: Create a list price field block
### Description

We should create a list price product field block to be displayed within the Basic details section.

#### Design

Image

AkVNGImLgSqCObTQ3idVn7-fi-205%3A21590

#### Acceptance criteria

* [ ] A list price block should be rendered in the product editor template within the basic details section (if the details section has been added already).
* [ ] The field should allow a user to change the list price and persist correctly
* [ ] The list price field should contain helper text with a link to the pricing settings screen.

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This creates a pricing block field that could be used for both the list price and the sale price.

Closes #37098 

Screenshot 2023-03-15 at 2 00 49 PM

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Build this branch and enable the product blocks feature flag: `block-editor-feature-enabled`
2. Go to **Products > Add new** and check if the pricing and sales field show up
3. Now create a product using the classic editor and add a price and sale price, load this product in the new product block form, the price and sale price should be displayed correctly.
4. Go to **WooCommerce > Settings** and change the currency
5. Reload the product block editor and make sure the icons in the price field match the new currency.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/product-editor/src/components/editor/init-blocks.ts [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 /**
  * Internal dependencies
  */
 import { init as initName } from '../details-name-block';
 import { init as initSection } from '../section';
 import { init as initTab } from '../tab';
+import { init as initPricing } from '../pricing-block';
 
 export const initBlocks = () => {
+	registerCoreBlocks();
```
Comment (louwie17) [inline packages/js/product-editor/src/components/editor/init-blocks.ts:15]: Needed this to get the `column` blocks, although it does look like some of the blocks are already being registered somewhere, although I can't figure out where.
Screenshot 2023-03-16 at 9 56 43 AM

File: packages/js/product-editor/src/components/pricing-block/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+						String( regularPrice ),
+						currencyConfig,
+						formatAmount
+					) }
+					{ ...inputProps }
+				/>
+			</BaseControl>
+			{ taxSettingsElement && (
+				<span className="woocommerce-product-form__secondary-text">
+					{ taxSettingsElement }
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/pricing-block/edit.tsx]: Why you are not using the `help` prop of the `BaseControl` component for this case?

File: packages/js/product-editor/src/components/pricing-block/edit.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	const { name, label, showPricingSection = false } = attributes;
+	const [ regularPrice, setRegularPrice ] = useEntityProp< string >(
+		'postType',
+		'product',
+		name
+	);
+	const context = useContext( CurrencyContext );
+	const { getCurrencyConfig, formatAmount } = context;
+	const currencyConfig = getCurrencyConfig();
+	const inputProps = useCurrencyInputProps( {
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/pricing-block/edit.tsx:37]: What do you suggest if we need to extends (not overrides) the `onFocus` or `onKeyUp` events coming from `useCurrencyInputProps`?

File: plugins/woocommerce-admin/client/products/product-block-page.scss [hunk 4]
```diff
@@ -0,0 +1,53 @@
+.woocommerce-product-block-editor {
```
Comment (mdperez86) [inline plugins/woocommerce-admin/client/products/product-block-page.scss:1]: I need this file for my summary block implementation.

**Review by mdperez86 — COMMENTED** [review]
Nice work here @louwie17. I left some small comments here.

**Comment by louwie17** [comment]
@mdperez86 this should be good for a re-review when you have time :)

File: packages/js/product-editor/src/components/pricing-block/edit.tsx (hunk 2 above)
Comment (louwie17) [reply packages/js/product-editor/src/components/pricing-block/edit.tsx]: Good point! Fixed here: https://github.com/woocommerce/woocommerce/pull/37211/commits/dfe22e54ed1b183a095ae9d053c265daac791824

File: packages/js/product-editor/src/components/pricing-block/edit.tsx (hunk 3 above)
Comment (louwie17) [reply packages/js/product-editor/src/components/pricing-block/edit.tsx:37]: Provided it as optional options: https://github.com/woocommerce/woocommerce/pull/37211/commits/5481c9a85233e5a217ced4947b24669649e7a640

**Review by mdperez86 — APPROVED** [review]
LGTM! Nice work here @louwie17. I will merge this because we need your changes for the resting blocks.

**Pull request merged**